compiler-core/src/type_/expression.rs

```diff
@@ -234,24 +234,28 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
         self.environment.new_unbound_var()
     }
 
-    pub fn infer(&mut self, expr: UntypedExpr) -> Result<TypedExpr, Error> {
+    pub fn infer(
+        &mut self,
+        expr: UntypedExpr,
+        errors: &mut Vec<Error>,
```

Comment:
How about we put `errors` on `self`? 

It's not specific to any one particular expr and every single function takes it as an argument, which makes me think it's an argument to the class as a whole, not to its methods. It would make this PR much smaller too!

Reply:
Yeah good call. Originally when I started I didn't realize it would have to be passed everywhere